content/en/blog/2023/dlb-connection-balance/index.md

```diff
@@ -0,0 +1,305 @@
+- Ordered: For multiple producers and consumers where the order of tasks is important. When multiple tasks are processed by multiple processor cores, they must be rearranged in the original order.
+- Atomic: For multiple producers and consumers, where tasks are grouped according to certain rules. These tasks are processed using the same set of resources and the order of tasks within the same group is important.
+
+An ingress gateway is expected to process as much data as possible as quickly as possible, so Intel DLB connection load balance uses an unordered queue.
+
+## How to use Intel DLB connection load balance in Istio
+
+With the 1.17 release, Istio supports Intel DLB connection load balance officially.
+
+The following steps show how to use Intel DLB connection load balance in an Istio [Ingress Gateway](../../../docs/tasks/traffic-management/ingress/ingress-control/) in an SPR (Sapphire Rapids) machine and Kubernetes is ready.
+
+### Step 1 Prepare DLB environment
+
+Install the Intel DLB driver by following [the instructions on the Intel DLB driver official site](https://www.intel.com/content/www/us/en/download/686372/intel-dynamic-load-balancer.html).
+
+Install the Intel DLB device plugin with the following command:
+
+{{< text bash >}}
+$ kubectl apply -k https://github.com/intel/intel-device-plugins-for-kubernetes/deployments/dlb_plugin?ref=v0.26.0
+{{< /text >}}
+
+For more details about the Intel DLB device plugin, please refer to [Intel DLB device plugin homepage](https://www.envoyproxy.io/docs/envoy/latest/configuration/other_features/dlb#config-connection-balance-dlb).
+
+You can check the Intel DLB device resource:
+
+{{< text bash >}}
+$ kubectl describe nodes | grep dlb.intel.com/pf
+  dlb.intel.com/pf:   2
+  dlb.intel.com/pf:   2
+...
+{{< /text >}}
+
+### Step 2 Download Istio
+
+Now the latest version is 1.17.2, let’s download the installation:
+
+{{< text bash >}}
+$ curl -L https://istio.io/downloadIstio | ISTIO_VERSION=1.17.2 TARGET_ARCH=x86_64 sh -
+$ cd istio-1.17.2
+$ export PATH=$PWD/bin:$PATH
+{{< /text >}}
+
+**TIPS: All following actions will be done under this directory.**
+
+You can check the version is 1.17.2:
+
+{{< text bash >}}
+$ istioctl version
+no running Istio pods in "istio-system"
+1.17.2
+{{< /text >}}
+
+### Step 3 Install Istio
+
+Create install configuration for Istio, notice that we assign 4 CPUs and 1 DLB device to ingress gateway and set concurrency as 4, which is equal to the CPU number.
+
+{{< text bash >}}
+$ cat > config.yaml << EOF
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  profile: default
+  components:
+    ingressGateways:
+    - enabled: true
+      name: istio-ingressgateway
+      k8s:
+        overlays:
+          - kind: Deployment
+            name: istio-ingressgateway
+            patches:
+              - path: spec.template.spec.containers.[name:istio-proxy].args.[-1]
+                value: "--concurrency=4"
```

Comment:
This is not the correct way to set concurrency, should use proxy config

Reply:
I use this to change, but not work:
```
apiVersion: networking.istio.io/v1beta1
kind: ProxyConfig
metadata:
  name: dlb-proxyconfig
  namespace: istio-system
spec:
  selector:
    matchLabels:
      app: istio-ingressgateway
  concurrency: 4
```

Reply:
From https://discuss.istio.io/t/understanding-the-concurrency-parameter-in-proxy-config/13737, it seems I should set 0 in proxyConfig?

Reply:
Check https://github.com/istio/istio/issues/23470#issuecomment-630302121, I still need to set it as `4`.

Reply:
@howardjohn could you help ensure my usage is correct?

Reply:
@howardjohn updated, please review again.